src/components/UserProfile/UserProfile.tsx

```
@@ -0,0 +1,500 @@
+}
+
+const UserProfile: React.FC<InterfaceUserProfileProps> = ({
+  userId,
+  isOwnProfile = false,
+  isAdminView = false,
+}) => {
+  const { t: tCommon } = useTranslation('common');
+  // Try to get location, fallback to null if Router context is missing (e.g., in tests)
+  let location: ReturnType<typeof useLocation> | null = null;
+  try {
+    location = useLocation();
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)